protocols/kad/src/behaviour/test.rs

```diff
@@ -695,6 +695,407 @@ fn put_record() {
+                },
+                None => panic!("Query not found: {:?}", qid),
+            }
+        }
+
+        // The accumulated results for one round of publishing.
+        let mut results = Vec::new();
+
+        block_on(poll_fn(move |ctx| {
+            // Poll all swarms until they are "Pending".
+            for swarm in &mut swarms {
+                loop {
+                    match swarm.poll_next_unpin(ctx) {
+                        Poll::Ready(Some(SwarmEvent::Behaviour(
+                            KademliaEvent::OutboundQueryCompleted {
+                                id,
+                                result: QueryResult::PutRecord(res),
+                                stats,
+                            },
+                        )))
+                        | Poll::Ready(Some(SwarmEvent::Behaviour(
+                            KademliaEvent::OutboundQueryCompleted {
+                                id,
+                                result: QueryResult::RepublishRecord(res),
+                                stats,
+                            },
+                        ))) => {
+                            assert!(qids.is_empty() || qids.remove(&id));
+                            assert!(stats.duration().is_some());
+                            assert!(stats.num_successes() >= replication_factor.get() as u32);
+                            assert!(stats.num_requests() >= stats.num_successes());
+                            assert_eq!(stats.num_failures(), 0);
+                            match res {
+                                Err(e) => panic!("{:?}", e),
+                                Ok(ok) => {
+                                    assert!(records.contains_key(&ok.key));
+                                    let record = swarm.behaviour_mut().store.get(&ok.key).unwrap();
+                                    results.push(record.into_owned());
+                                }
+                            }
+                        }
+                        Poll::Ready(Some(SwarmEvent::Behaviour(
+                            KademliaEvent::InboundPutRecordRequest { record: _, .. },
+                        ))) => {
+                            // Ignore the record
+                        }
+                        // Ignore any other event.
+                        Poll::Ready(Some(_)) => (),
+                        e @ Poll::Ready(_) => panic!("Unexpected return value: {:?}", e),
+                        Poll::Pending => break,
+                    }
+                }
+            }
+
+            // All swarms are Pending and not enough results have been collected
+            // so far, thus wait to be polled again for further progress.
+            if results.len() != records.len() {
+                return Poll::Pending;
+            }
+
+            // Consume the results, checking that each record was replicated
+            // correctly to the closest peers to the key.
```

Comment:
This is outdated, right?

Reply:
Not any more with the new approach, I think? Or do you mean to rewrite the comment itself?